docs/source/howto/codes.rst

```diff
@@ -45,8 +45,84 @@ Design guidelines
 Running external codes
 ======================
 
-`#3987`_
+To run an external code with AiiDA, you will need to use an appropriate :ref:`calculation plugin <topics:plugins>` that knows how to take transform the input nodes into the input files that the code expects, copy everything in the code's machine, run the calculation and retrieve the results.
+This plugin would have already been created, either by you (see section on :ref:`interfacing external codes <how-to:codes:plugin>`) or by other AiiDA plugin-developers (available packages are listed in the `plugin registry <https://aiidateam.github.io/aiida-registry/>`_), and has been properly installed in your environment.
+You can check which ones you have currently available by running the ``verdi plugin`` command:
 
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations
+
+For the purpose of this example, we will use the ``arithmetic.add`` plugin, which is a plugin for bash to sum two integers that comes pre-installed with ``aiida-core``.
+You can access it through the use of the ``CalculationFactory``:
+
+.. code-block:: python
+
+    from aiida.plugins import CalculationFactory
+    calculation_class = CalculationFactory('arithmetic.add')
+
+We will obviously also need to provide the inputs for the code when running the calculation.
+You can also use ``verdi plugin`` to check this and other relevant information for specific plugins:
+
+.. code-block:: bash
+
+    $ verdi plugin list aiida.calculations arithmetic.add
+
+You will see that 3 inputs nodes are required: two containing the values to add up (``x``, ``y``) and one containing information about the specific code to execute (``code``).
+If you already have these nodes in your database, you can get them by :ref:`querying for them <how-to:data:finding-data>` or using ``orm.load_node(<PK>)``.
+Otherwise, you will need to create them as shown below (note that you `will` need to already have the ``localhost`` computer configured):
+
+.. code-block:: python
+
+    from aiida import orm
+    code_node = orm.Code(remote_computer_exec=[localhost, '/bin/bash'])
+    numx_node = orm.Int(17)
+    numy_node = orm.Int(11)
+
+To provide these as inputs to the calculations, we will now use the ``builder`` object that we can get from the class:
+
+.. code-block:: python
+
+    calculation_builder = calculation_class.get_builder()
+    calculation_builder.code = code_node
+    calculation_builder.x = numx_node
+    calculation_builder.y = numy_node
+
+Now everything is in place and ready to perform the calculation, which can be done in two different ways.
```

Comment:
I'm wondering if we can hand-off to a different section here. How to run and submit calculations is already described in a different how-to.

Reply:
Wait, what? Where? I thougth this was the how-to on running calculations.

Reply:
Honestly, I might have gotten confused here. Let's leave this and I'll double check any potential duplication after everything is merged.